### ISSUE TITLE
Data augmentation layers generating multiple images

### ISSUE THREAD

**Issue by FalsoMoralista** [GoogleCloudPlatform/practical-ml-vision-book#1]
@lakshmanok, i have recently read you article on writing a custom augmentation layer ([this tutorial](https://github.com/GoogleCloudPlatform/practical-ml-vision-book/blob/master/06_preprocessing/06e_colordistortion.ipynb)) and wondered if you have any ideas on how to handle augmentation layers that generates multiple images. I have implemented one model to do so but i don't know how to associate the original labels to the corresponding generated images which leads to mismatches between logits and labels when training the model. 

Code in the link below:
 https://stackoverflow.com/questions/68136779/logits-and-labels-must-be-broadcastable-data-augmentation-layers-makes-logits-a

Any help and suggestions are welcome.

**Comment by lakshmanok** [comment]
Any Keras layer will have to keep the batch size the same as the input, so it's not possible as a Keras layer.

If you really want to generate multiple images, you would have to do this in the ingest pipeline.

That said, the more common approach is to randomly select one out of the multiple images during each epoch. If you do that, you can keep it as a layer within the model.

**Comment by FalsoMoralista** [comment]
I understand, thank you very much for your time.
